tower-http/src/lib.rs

```diff
@@ -1,3 +1,114 @@
+//! `async fn(HttpRequest) -> Result<HttpResponse, Error>`
+//!
+//! # Overview
+//!
+//! `tower-http` is a library that provides HTTP-specific middlewares and utilities built on top of
+//! the [`tower`] and [`http`] crates.
+//!
+//! All middlewares uses the [`http`] and [`http-body`] crates as the HTTP abstractions. That means
+//! they're compatible with any library or framework that also uses those crates, such as
+//! [`hyper`].
+//!
+//! # Example server
+//!
+//! This example shows how to apply middlewares from `tower-http` to a [`Service`] and then run
+//! that service using [`hyper`].
+//!
+//! ```rust,no_run
```

Comment:
alternatively we could avoid the `no_run` by just sticking the body of the `main` fn inside another function, hiding that function's definition using `#`, and never calling it. then, the docs won't say "this example is not run". :)

Reply:
Is that a docs.rs thing because it doesn't say so when building the docs locally 🤔